installutils/kubeinstall/kube_installer_test.go

```diff
@@ -52,29 +56,48 @@ var _ = Describe("KubeInstaller", func() {
 			kube.WaitForNamespaceTeardown(ns)
 		}
 		ns = "test" + testutils.RandString(5)
-		testutils.SetupKubeForTest(ns)
+		err := kubeutils.CreateNamespacesInParallel(kubeClient, ns)
+		Expect(err).NotTo(HaveOccurred())
+
+		idPrefix := fmt.Sprintf("supergloo-helm-%s-%d", os.Getenv("BUILD_ID"), config.GinkgoConfig.ParallelNode)
+		lock, err = clusterlock.NewTestClusterLocker(kube.MustKubeClient(), clusterlock.Options{
+			IdPrefix: idPrefix,
+		})
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(lock.AcquireLock()).NotTo(HaveOccurred())
 	})
 	AfterEach(func() {
-		testutils.TeardownKube(ns)
+		err := kubeutils.DeleteNamespacesInParallelBlocking(kubeClient, ns)
+		Expect(err).NotTo(HaveOccurred())
 		kube.TeardownClusterResourcesWithPrefix(kubeClient, "istio")
 		kube.TeardownClusterResourcesWithPrefix(kubeClient, "prometheus")
 		kube.WaitForNamespaceTeardown(ns)
+		Expect(lock.ReleaseLock()).NotTo(HaveOccurred())
 	})
+	// TODO(EItanya): rewrite this test so that it uses a non-
 	Context("updating resource from cache", func() {
 		It("does nothing if the resource hasnt changed", func() {
-
+			ownerLabels := map[string]string{
+				"unique": "label",
```

Comment:
should use a named variable for this key and value since it's used below as well, could generate it randomly to "ensure" uniqueness, actually, a util for getting a random unique label would be useful